pkg/controller/daemon/daemoncontroller.go

```diff
@@ -580,7 +583,15 @@ func (dsc *DaemonSetsController) syncNodes(ds *extensions.DaemonSet, podsToDelet
 	for i := 0; i < createDiff; i++ {
 		go func(ix int) {
 			defer createWait.Done()
-			if err := dsc.podControl.CreatePodsOnNode(nodesNeedingDaemonPods[ix], ds.Namespace, &template, ds); err != nil {
+			isController := true
```

Comment:
Could you set the BlockOwnerDeletion like what we did for replica set controller [here](https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/replicaset/replica_set.go#L490-L491)? Please update all for all the controllers.